inskin-media/inskin-media-htb.js

```diff
@@ -0,0 +1,564 @@
+     *
+     * @param {object[]} returnParcels The array of original parcels, SAME array that was passed to
+     * generateRequestObj to signal which slots need demand. In this funciton, the demand needs to be
+     * attached to each one of the objects for which the demand was originally requested for.
+     */
+    function __parseResponse(sessionId, adResponse, returnParcels) {
+        /* =============================================================================
+         * STEP 4  | Parse & store demand response
+         * -----------------------------------------------------------------------------
+         *
+         * Fill the below variables with information about the bid from the partner, using
+         * the adResponse variable that contains your module adResponse.
+         */
+
+        /* This an array of all the bids in your response that will be iterated over below. Each of
+         * these will be mapped back to a returnParcel object using some criteria explained below.
+         * The following variables will also be parsed and attached to that returnParcel object as
+         * returned demand.
+         *
+         * Use the adResponse variable to extract your bid information and insert it into the
+         * bids array. Each element in the bids array should represent a single bid and should
+         * match up to a single element from the returnParcel array.
+         *
+         */
+
+        /* ---------- Process adResponse and extract the bids into the bids array ------------ */
+
+        /* --------------------------------------------------------------------------------- */
+
+        var hasBids = false;
+        var bidsMap = {};
+
+        for (var j = 0; j < returnParcels.length; j++) {
+            var curReturnParcel = returnParcels[j];
+
+            var headerStatsInfo = {};
+            var htSlotId = curReturnParcel.htSlot.getId();
+            headerStatsInfo[htSlotId] = {};
+            headerStatsInfo[htSlotId][curReturnParcel.requestId] = [curReturnParcel.xSlotName];
+
+            var curBid = adResponse.decisions[curReturnParcel.xSlotName];
+
+            /* No matching bid found so its a pass */
+            if (!curBid) {
+                if (__profile.enabledAnalytics.requestTime) {
+                    __baseClass._emitStatsEvent(sessionId, 'hs_slot_pass', headerStatsInfo);
+                }
+                curReturnParcel.pass = true;
+
+                continue;
+            }
+
+            /* ---------- Fill the bid variables with data from the bid response here. ------------ */
+
+            /* Using the above variable, curBid, extract various information about the bid and assign it to
+             * these local variables */
+
+            /* The bid price for the given slot */
+            var bidPrice = curBid.pricing.clearPrice;
+
+            /* The size of the given slot */
+            var bidSize = [Number(curBid.width), Number(curBid.height)];
```

Comment:
The bid response example above does not contain information of `width` or `height`. Please confirm the format of the bid response.